src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,55 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    public static int splitAndSum(String text) {
+        if (isBlank(text)) {
+            return 0;
+        }
+
+        return sum(toInt(split(text)));
+    }
+
+    private static boolean isBlank(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static int[] toInt(String[] values) {
+        int[] numbers = new int[values.length];
+
+        for (int i = 0; i < values.length; i++) {
+            numbers[i] = toPositive(values[i]);
+        }
+
+        return numbers;
+    }
+
+    private static int toPositive(String value) {
+        int number = Integer.parseInt(value);
+        if (number < 0 ) {
+            throw new RuntimeException("입력에 음수가 포함될 수 없습니다.");
+        }
+
+        return number;
+    }
+
+    private static String[] split(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            return m.group(2).split(customDelimiter);
+        }
+
+        return text.split(",|:");
+    }
+
+    private static int sum(int[] numbers) {
+        int sum = 0;
+        for (int number : numbers) {
+            sum += number;
+        }
+
+        return sum;
```

Comment:
```suggestion
        return Arrays.stream(numbers)
                .sum();
```

Stream API를 사용하여 반복문을 내부 라이브러리에 위임함으로써 무엇을 연산할지를 드러내는 것은 어떨까요

Reply:
이 부분은 수업에서 자바지기님께서 객체지향을 훈련할 땐
스트림도 '지양'하란 말씀이 있으셔서 일단은 찝찝하지만 for문으로 했습니다.

저도 사실 js를 주로 써서 함수형의 표현력을 사랑하는데 말이죠ㅠ.ㅠ